### PR TITLE
[@types/amphtml-validator] New type definitions

### PR DESCRIPTION
**Pull request by kevincharm**
Hello! Submitting a new type definition for [`amphtml-validator`](https://www.npmjs.com/package/amphtml-validator). Please let me know if any changes are needed. Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/amphtml-validator/tslint.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (andy-ms) [inline types/amphtml-validator/tslint.json]: Please just use the default settings (entire file should be `{ "extends": "dtslint/dt.json" }`) -- I don't like semicolons either but it's better for the repository to stay consistent.

File: types/amphtml-validator/amphtml-validator-tests.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as ampHtmlValidator from 'amphtml-validator'
+
+;(async () => {
+    const validator = await ampHtmlValidator.getInstance()
+    const result = validator.validateString('<html></html>')
```
Comment (andy-ms) [inline types/amphtml-validator/amphtml-validator-tests.ts]: Tests should include a use of `result` too

File: types/amphtml-validator/validator.proto.d.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+ */
+
+export type ValidationResultStatus =
+    | 'UNKNOWN'
+    | 'PASS'
+    | 'FAIL'
+
+export type ValidationErrorSeverity =
+    | 'UNKNOWN_SEVERITY'
+    // A document with at least one error of this severity fails validation.
```
Comment (andy-ms) [inline types/amphtml-validator/validator.proto.d.ts]: I think it's more common to put the comment above the thing it comments? It's obvious what you're doing here, less so if you're in the middle of the long list of `ErrorCategoryCode`.

File: types/amphtml-validator/validator.proto.d.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
+/**
```
Comment (andy-ms) [inline types/amphtml-validator/validator.proto.d.ts]: This file contains only type definitions -- might be better to just include all of this in `index.d.ts`.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by andy-ms** [comment]
@kevincharm Please address comments from the code reviewers.

File: types/amphtml-validator/tslint.json (hunk 1 above)
Comment (kevincharm) [reply types/amphtml-validator/tslint.json]: Updated tslint & re-linted the new code.

File: types/amphtml-validator/amphtml-validator-tests.ts (hunk 2 above)
Comment (kevincharm) [reply types/amphtml-validator/amphtml-validator-tests.ts]: Added some usage of the `result` object and its members `status` and `errors` as well.

File: types/amphtml-validator/validator.proto.d.ts (hunk 3 above)
Comment (kevincharm) [reply types/amphtml-validator/validator.proto.d.ts]: These were copied verbatim from the original codebase (a protobuf file). I've just removed all the comments.

File: types/amphtml-validator/validator.proto.d.ts (hunk 4 above)
Comment (kevincharm) [reply types/amphtml-validator/validator.proto.d.ts]: I've consolidated the extra type definitions all into `index.d.ts`.

**Comment by kevincharm** [comment]
@andy-ms thanks for the review! I've addressed your comments and pushed a new commit. Let me know if there's anything else. Cheers.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**